manuscript/manuscript.Rmd

```diff
@@ -122,27 +122,33 @@ The goal of the agent is to learn how to maximize its expected cumulative reward
+Conversely, *Model-based* algorithms either learn or use a model of the environment to achieve the RL objective.
 A trade-off between these approaches is that when it is possible to quickly learn a model of the environment or the model is already known, model-based algorithms tend to require much less interaction with the environment to learn good-performing policies [@mbpo].
 Yet, frequently, learning a model of the environment is very difficult, and in these cases, model-free algorithms tend to outperform [@CITE].
 Since most of the recent focus in RL has gone towards model-free algorithms, we will focus on model-free agents for the rest of the paper.
 
-Amongst model-free algorithms, there is divergence around what functions the agent is attempting to learn.
+Among model-free algorithms, there is divergence around what functions the agent is attempting to learn in order to achieve the RL objective.
 Generally, model-free agents are either learning a policy function, a value function or both.
 For context, *value functions* are proxies for how high of a cumulative reward an agent can expect to receive from a given state or state-action pair.
 If an agent knows the value function, then the agent can find the optimal action at any state by selecting the action that will maximize the value function in expectation.
 The class of algorithms that exclusively try to learn a value function are called *value learning* methods.
 *Policy gradient* algorithms exclusively learn the policy.
-*Actor-critic* algorithms attempt to learn both a policy function and a value function, whereby the value function, which is learned separately from the policy, is used to inform the agent on the goodness of a selected action.
-There is a handful of advantages and disadvantages for each of these approaches that we will not go in depth -- for more, see [@CITE]; yet, the main point of distinction comes from the bias-variance trade-off.
-Policy gradients tend to struggle with high variance, value-learning tends to be biased while actor-critic methods balance these two concerns.
-
-
-
-
+And, lastly, *actor-critic* algorithms attempt to learn both a policy function and a value function, whereby the value function, which is learned separately from the policy, is used to inform the agent on the goodness of a selected action.
+There is a handful of advantages and disadvantages for each of these approaches that we will not go in depth on -- for more, see [@CITE]; yet, the main point of divergence comes from the bias-variance trade-off [@CITE].
+Generally, policy gradient algorithms tend to struggle with high variance, value-based learning tends to be biased while actor-critic methods balance these two concerns [@CITE; @CITE; @CITE].
+
+Neural networks become useful in RL when the environment has a large state-action space, which happens frequently with realistic decision-making problems. 
```

Comment:
see above, I think all of this can move down to the next section?

We might want a one sentence here just saying that neural networks are generic function approximators and can be used to approximate various functions the algorithm may be trying to learn, but hold off on a longer discussion until later.  (or just condense this last paragraph into one sentence that algorithms using neural networks to approximate learned functions have been particularly successful in large state-action spaces where other methods have failed).

Reply:
@cboettig So incorporating suggestions from this and above, are you thinking about having a paragraph that's kinda like:
-Generally describe the classic approaches to solving RL problems (DP and TD learning)
-Explain why these fail for most realistic problems
-give a sentence or 2 on neural nets and how they can overcome issues with classic methods
?

Reply:
Yeah, I think that's good, though not sure how much detail we need to get into. We can probably get away with merely acknowledging rather than actually explaining classic approaches.  The Marescot 2013 review is a nice overview of DP, and that can cite that recent Briar and Munch paper as an ecological example of TD.